docs/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,104 @@
+
+Now we need to change the Workflow for some reason, so we add `2.0` to the graph as the new default:
+
+![2.0 Graph](/img/worker-versioning/graph-prog-2.svg)
+
+All new Workflows started on the queue will have their first tasks assigned to version `2.0`. Existing `1.0` Workflows will keep generating tasks targeting `1.0`. Each deployment of Workers will receive their respective tasks. This same concept carries forward for each new incompatible version.
+
+Maybe we have a bug in `2.0`, and we want to make sure all open `2.0` workflows switch over to some new code as fast as possible. So we add `2.1` to the graph, marking it as compatible with `2.0`. Now our graph looks like this:
+
+![2.1 Graph](/img/worker-versioning/graph-prog-2.1.svg)
+
+All new Workflow Tasks which are generated for Workflows whose last Workflow Task completion was on version `2.0` will now be assigned to version `2.1`. Because we specified that `2.1` is compatible with `2.0`, it is assumed that workers with this version will be able to process these existing histories successfully.
+
+We continue with our normal development cycle, adding a `3.0` version. Nothing new here:
+
+![3 Graph](/img/worker-versioning/graph-prog-3.svg)
+
+Now imagine that `3.0` doesn't have an explicit bug, but there's something about the business logic we find less than ideal. We are OK with existing `3.0` workflows running to completion, but we would like new workflows to go back to the old `2.x` branch. This operation is supported by performing an update targeting `2.1` and setting it as the current default, which will result in this graph:
+
+![Reshuffled Graph](/img/worker-versioning/graph-prog-reshuffle.svg)
+
+Now new workflows will start on `2.1`.
+
+#### Permitted and forbidden operations on the graph
+
+A request to change the graph can do one of the following:
+
+1. Add a new version to the graph, as a default version
+2. Add a new version to the graph, compatible with some existing version.
+3. Add a new version to the graph, compatible with some existing version and as the new default.
+4. Reorder an existing version (and it's whole compatible branch). We allow moving a node in the incompatible line
+   to the front (along with its entire compatible branch), as long as you have targeted the most recent
+   version in that compatible branch.
+
+You can't delete versions. This is to help avoid a situation where workflows accidentally become stuck with
+no means of making progress because the version they are associated with no longer exists.
+
+Sometimes you might want to do this intentionally. If you _want_ to make sure that all workflows currently
+being processed by, say, `2.0` stop, even if you do not yet have a new version ready, you can add a new
+version `2.1` to the graph marked as compatible with `2.0`. New tasks will target `2.1`, but since you
+have not yet deployed any `2.1` workers, they will not make any progress.
```

Comment:
```suggestion
However, sometimes you might want to do this intentionally.
If you _want_ to make sure that all Workflows currently being processed by, say, `2.0` stop, even if you don't yet have a new version ready, you can add a new version `2.1` to the graph marked as compatible with `2.0`.
New tasks will target `2.1`, but since you have not yet deployed any `2.1` workers, they will not make any progress.
```